core-bundle/src/EventListener/SearchIndexListener.php

```diff
@@ -58,19 +59,28 @@ public function __invoke(TerminateEvent $event): void
 
         $document = Document::createFromRequestResponse($request, $response);
         $needsIndex = $this->needsIndex($request, $response, $document);
-        $success = $event->getResponse()->isSuccessful();
+        $needsDelete = $this->needsDelete($response, $document);
```

Comment:
@fritzmg Can you please confirm that it is correct to remove the `$success` variable and to add a try/catch block in line 64?